scripts/upload_test_results/upload_to_github.py

```diff
@@ -2,168 +2,160 @@
+        # Use created repo to store results
+        origin = Repo(os.path.join(repo_dir))
+        upstream = Repo(os.path.join(repo_dir))
+        origin.git.checkout('master')
+        logging.info('Pull updates from the upstream ...')
+        upstream.git.pull()
+        logging.info('Push updates to the origin ...')
+        origin.git.push(origin_name, 'master')
+
+        try:
+            # Create a new branch for daily master results
+            new_branch_name = branch_git_title + commit_name
+            origin.create_head(new_branch_name)
+            logging.info('Checkout to the new branch %s', new_branch_name)
+            origin.git.checkout(new_branch_name)
+        except Exception:
+            logging.info('Branch for the current commit already existing, skip step')
+
+    # create full git directory path where your results will be stored
+    git_dir_path = os.path.join(repo_dir, 'results', ver_folder_name)
+    logging.info('Destination directory path to store file in Git %s', git_dir_path)
+
+    if not os.path.exists(git_dir_path):
+        os.mkdir(git_dir_path)
+        logging.info('Path created %s', git_dir_path)
+    else:
+        logging.info('Path already existing')
+
+    return git_dir_path, new_branch_name, origin
+
+def add_push_to_github(git_dir_path, origin, commit_msg, title_msg):
+    # Add your test results to the repo, commit them and then push them to create PR on Github
+    for j in glob.glob("{}/*.xml".format(file_path)):
+        file_name = os.path.basename((j))
+        logging.info("File name %s", file_name)
+
+        # Return path to its initial status
+        source_file_path = argument.res_dir
+        source_file_path = os.path.join(source_file_path, file_name)
+        logging.info('Source directory file path %s', source_file_path)
+        git_file_path = os.path.join(git_dir_path, file_name)
+        logging.info('Destination directory path file to commit %s', git_file_path)
+        shutil.copy(source_file_path, git_file_path)
+        origin.index.add(git_file_path)
     try:
-        ver_url = 'https://testing.zephyrproject.org/daily_tests/versions.json'
-        ver_file = wget.download(ver_url, 'versions.json')
-    except:
-        print("\nERROR: Can't download versions.json file. Script execution ended.")
+        logging.info('Commit changes ...')
+        author = Actor(author_name, author_email)
+        origin.git.commit('-m', ver_folder_name + ": " + company_name + " "+ title_msg + \
+        "\n\n" + commit_msg + \
+        "\n\nSigned-off-by: " + author_name + " " + author_email, author=author)
+        logging.info('Push commit to the origin ...')
+        origin.git.push('origin', new_branch_name)
+        logging.info('Success! Go to your GitHub and create PR to merge the new origin branch with the upstream')
+        origin.git.checkout('master')
+
+    except Exception:
+        logging.critical('ERROR: Nothing to commit.')
```

Comment:
I think there can be other errors causing git.push to fail, but for the early version of the script I don't mind. It will be easier to modify the script after we start working with it and seeing real issues.